packages/autorest.python/autorest/codegen/models/primitive_types.py

```diff
@@ -608,3 +608,28 @@ def get_declaration(self, value: str) -> str:
     @property
     def instance_check_template(self) -> str:
         return "isinstance({}, bytes)"
+
+
+class AzureCoreType(PrimitiveType):
```

Comment:
i don't think this would be consistent with the azure core model design that @tadelesh did. What is the status of that design @tadelesh ? and would we be able to treat these the same way?

Reply:
Do you mean the referenced type? Since we do not have a final decision for the tsp referenced type yet, the design does not have much progress. I think it's ok to introduce this `AzureCoreType`, once we decide to add referenced type, I could consolidate the two models.